### PR TITLE
docs: updated release instructions

### PR DESCRIPTION
**Pull request by vvagaytsev**
**What this PR does / why we need it**:
This PR adds some release-related corrections to the contributing guide.

### CODE REVIEW

**Comment by vvagaytsev** [comment]
This is ready for review now.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -276,7 +276,8 @@ To make a new release, set your current working directory to the garden root dir
 2. The next step depends on the release type:
     * If you're making the first pre-release:
         1. Reset `latest-release` to `master` with `git reset --hard origin/master`.
-        2. Run `./scripts/release.ts preminor|prepatch`.
+        2. Run `git log` to make sure that the latest commit is the expected one and there are no unwanted changes from master included into the release.
```
Comment (thsig) [inline CONTRIBUTING.md:279]: "included into" -> "included in"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -286,14 +287,16 @@ To make a new release, set your current working directory to the garden root dir
         3. Run `./scripts/release.ts minor | patch`. This way, the version bump commits and changelog entries created by the pre-releases are omitted from the final history.
 3. If you're making a pre-release you're done, and you can now start testing the binaries that were just published to our Github [Releases page](https://github.com/garden-io/garden/releases) (**step 4**). Otherwise, go to **step 5**.
 4. Manual testing (using the pre-release/release binary)
-    * On a **Windows** machine, run `garden dev --hot=vote` in the `vote` example project.
+    * On a **Windows** machine, run `garden deploy --dev vote --env remote` in the `vote` example project.
+        * If there are any issues with syncing, consider changing the `services[].devMode.sync[].mode` value(s) to `one-way-replica` and restart the Garden.
```
Comment (thsig) [inline CONTRIBUTING.md:291]: "restart the" -> "restart"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -286,14 +287,16 @@ To make a new release, set your current working directory to the garden root dir
...
         * Change a file in the `vote` service and verify that the hot reload was successful.
         * Open the dashboard, verify that the initial page loads without errors.
-    * On **macOS** or **Linux**, run the `./scripts/test-release <version>` script. The script runs some simple tests to sanity check the release.
+    * On **macOS** or **Linux**, run the `./scripts/test-release.sh <version>` script, where the `<version>` should be in the format of `<major>.<minor>.<patch>-<preReleaseCounter>`, e.g. `0.12.38-0`. The script runs some simple tests to sanity check the release.
```
Comment (thsig) [inline CONTRIBUTING.md:294]: "where the" -> "where"

"be in the format of" -> "have the format"

**Review by thsig — CHANGES_REQUESTED** [review]
Thanks for the edits! Had a few text-related comments there.

**Comment by vvagaytsev** [comment]
CR-based fixes have been done, @thsig could you review it again, please?

**Review by thsig — APPROVED** [review]

**Pull request merged**